sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/_sip_routing_client.py

```diff
@@ -0,0 +1,251 @@
+
+    :param endpoint: The endpoint url for Azure Communication Service resource.
+    :type endpoint: str
+    :param credential: The credentials with which to authenticate.
+    :type credential: TokenCredential
+    :keyword api_version: Api Version. Default value is "2021-05-01-preview". Note that overriding
+     this default value may result in unsupported behavior.
+    :paramtype api_version: str
+    """
+
+    def __init__(
+        self,
+        endpoint,  # type: str
+        credential,  # type: TokenCredential
+        api_version="2021-05-01-preview", # type: str
+        **kwargs  # type: Any
+    ):  # type: (...) -> SipRoutingClient
+
+        if not credential:
+            raise ValueError("credential can not be None")
+        try:
+            if not endpoint.lower().startswith("http"):
+                endpoint = "https://" + endpoint
+        except AttributeError:
+            raise ValueError("Host URL must be a string")
+
+        parsed_url = urlparse(endpoint.rstrip("/"))
+        if not parsed_url.netloc:
+            raise ValueError("Invalid URL: {}".format(endpoint))
+
+        self._endpoint = endpoint
+        self._authentication_policy = get_authentication_policy(endpoint, credential)
+
+        self._rest_service = SIPRoutingService(
+            self._endpoint,
+            authentication_policy=self._authentication_policy,
+            sdk_moniker=SDK_MONIKER,
+            api_version=api_version,
+            **kwargs
+        )
+
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str,  # type: str
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> SipRoutingClient
+        """Factory method for creating client from connection string.
+
+        :param str conn_str: Connection string containing endpoint and credentials.
+        :returns: The newly created client.
+        :rtype: ~azure.communication.siprouting.SipRoutingClient
+        """
+        endpoint, credential = parse_connection_str(conn_str)
+        return cls(endpoint, credential, **kwargs)
+
+    @distributed_trace
+    def get_trunk(
+        self,
+        trunk_fqdn,  # type: str
+        **kwargs  # type: Any
+    ):  # type: (...) -> SipTrunk
+        """Getter for single SIP trunk.
+
+        :param trunk_fqdn: FQDN of the desired SIP trunk.
+        :type trunk_fqdn: str
+        :returns: SIP trunk with specified trunk_fqdn.
+        :rtype: ~azure.communication.siprouting.models.SipTrunk
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError, LookupError
+        """
+        if trunk_fqdn is None:
+            raise ValueError("Parameter 'trunk_fqdn' must not be None.")
+
+        config = self._rest_service.get_sip_configuration(
+            **kwargs)
+        trunk = config.trunks[trunk_fqdn]
+
+        if not trunk:
+            raise LookupError("No entry found for FQDN:" + trunk_fqdn)
```

Comment:
I don't think we should be raising a `LookupError` here.
I'm not really a fan of raising in this scenario. Is that what's being done across languages?

Reply:
Changed. Looking at other languages, it makes the most sense to just return None, in case it's not found.